lib/std/crypto.zig

```diff
@@ -1,55 +1,65 @@
-pub const Md5 = @import("crypto/md5.zig").Md5;
-pub const Sha1 = @import("crypto/sha1.zig").Sha1;
-
-const sha2 = @import("crypto/sha2.zig");
-pub const Sha224 = sha2.Sha224;
-pub const Sha256 = sha2.Sha256;
-pub const Sha384 = sha2.Sha384;
-pub const Sha512 = sha2.Sha512;
-
-const sha3 = @import("crypto/sha3.zig");
-pub const Sha3_224 = sha3.Sha3_224;
-pub const Sha3_256 = sha3.Sha3_256;
-pub const Sha3_384 = sha3.Sha3_384;
-pub const Sha3_512 = sha3.Sha3_512;
+/// Hash functions.
+pub const hash = struct {
+    /// Legacy hash functions, that shouldn't be used in new applications.
+    pub const legacy = struct {
+        pub const Md5 = @import("crypto/md5.zig").Md5;
+        pub const Sha1 = @import("crypto/sha1.zig").Sha1;
+    };
```

Comment:
The intent was to make it clear that they should be avoided in new application, and that they may eventually be removed from the standard library.

Reply:
> they should be avoided in new application

It can be doc's comment.

Reply:
In addition, when they will be removed/deprecated, we can simply do this trick which is favoured by the Zig's stdlib:

```zig
pub const Md5 = @compileError("Md5 has been removed; use Sha3 instead");
```

or some other compile error message.

Reply:
>they will be removed/deprecated

Michel de Nostredame wrote (quatrain MDV): "They will be used for a very long time".

Reply:
As expected, Zig already has everything needed to manage deprecation 👍 

@kubkon Are you also suggesting that these primitives should be moved back with others instead of being in a `legacy` namespace?

I'd rather depend on warnings in documentation for what cannot be expressed in an API, but I'd be happy to revert that change if you also feel like this unnecessary.

Reply:
Oh sure, I didn't suggest we deprecate them now or in the future per se. I just wanted to point the mechanism that *is* used for deprecating public API in Zig. Sorry for confusion!

I do agree with @data-man and @andrewrk that putting `Md5` and `Sha1` in `legacy` is not necessary as long as we clearly point out in the docs they are cryptographically weak and should be generally avoided. However, if the user wants to use them, it's their responsibility to work out the dangers, etc.